File: python/ccxtpro/test/exchange/test_watch_ticker.py
# -*- coding: utf-8 -*-

from exchange.test_ticker import test_ticker
from ccxt import NetworkError


async def test_watch_ticker(exchange, symbol):
    # todo add real ticker tests here
    # log (symbol.green, 'watching order book...')
    method = 'watchTicker'
    # we have to skip some exchanges here due to the frequency of trading
    skipped_exchanges = [
        'ripio',
        'mexc'
    ]
    if exchange.id in skipped_exchanges:
        print(exchange.id, method, 'test skipped')
        return
    if (method in exchange.has) and exchange.has[method]:
        response = None
        now = exchange.milliseconds()
        end = now + 10000
        while now < end:
            try:
                response = await getattr(exchange, method)(symbol)
                now = exchange.milliseconds()
                test_ticker(exchange, response, method, symbol)
            except NetworkError:
                now = exchange.milliseconds()
        return response
    else:
        print(exchange.id, method + '() is not supported or not implemented yet')


__all__ = ['test_watch_ticker']

File: php/pro/test/Exchange/test_watch_ticker.php
<?php

function test_watch_ticker($exchange, $symbol) {

    echo __FUNCTION__ . "\n";

    $method = 'watchTicker';

    // we have to skip some exchanges here due to the frequency of trading
    $skipped_exchanges = array(
        'cex',
        'ripio',
        'mexc',
        'woo',
        'alpaca'
    );

    if (in_array($exchange->id, $skipped_exchanges)) {

        echo $exchange->id, ' ', $method, "() test skipped\n";
        return;
    }

    if (!(array_key_exists($method, $exchange->has) && $exchange->has[$method])) {
        echo $exchange->id, ' ', $method, "() is not supported or not implemented yet\n";
        return;
    }

    $now = $exchange->milliseconds();
    $ends = $now + 10000;
    while ($now < $ends) {
        try {
            $ticker = yield $exchange->watch_ticker($symbol);
            echo $ticker['datetime'], ' ', $exchange->id, ' ', $symbol, ' watch_ticker ', $ticker['last'] . "\n";
        } catch (Exception $e) {
            if (!($e instanceof \ccxt\NetworkError)) {
                throw $e;
            }
        }
        $now = $exchange->milliseconds();
    }
}
